ON-42104 # Added developer key and app association documentation (#505)
oneblink/sdk-node-js: DOCUMENTATION.md

import * as Types from '@oneblink/sdk/types'
```

## Permissions

The majority of actions in this SDK require that your developer key has particular permissions. These permissions can be set in your tenant's console. If you do not have access to your tenant's console, you will need to work with your Administrator to associate the permissions that you require with your key. The associated permissions for each function are detailed on the function definitions.

### Developer Key & Role

Actions that require a developer key with a minimum role permission will have the following in the function definition:

**Minimum Role Permission**

> The developer key used must be assigned a role with at least the permission documented for each action. E.g. If action's minimum role permission is _Forms: Read Only_, the role assigned to the developer key could have _Forms: Read Only_ or _Forms: Manager_.

Reviewer: ```suggestion
The developer key used must be assigned a role with at least the permission documented for each action. E.g. If the action's minimum role permission is _Forms: Read Only_, the role assigned to the developer key could have _Forms: Read Only_ or _Forms: Manager_.
```

### Developer Key & App Association

Actions that require a developer key to be assigned to an app will have the following in the function definition:

**App Association Required**

> These actions require the developer key to be associated with an app that is associated with the form that is being actioned. A form can be associated with an app in the following ways:

Reviewer 2: ```suggestion
In the case of any actions relating to forms, the assigned app must be associated with the form that is being actioned. A form can be associated with an app in the following ways:
```

Author: This implies that not all of these actions relate to forms which is incorrect. They are all associated with a single form.

Reviewer 2: Yeh but that might not always be the case. And tying the two concepts together is more for the user to understand IMO.

Reviewer 2: I'm looking at it this way: Currently all the actions that require an app association just happen to be single form actions.

> - For _Forms List_ type apps, the form must be in the Forms List menu item

Reviewer: ```suggestion
- For _Forms List_ type apps, the form must be in the Forms List menu item.
```

> - For _Forms List_ type apps, the form must be in the Forms List menu item

Reviewer 2: ```suggestion
- For _Forms List_ type apps, the form can be in the _Forms List_ menu item.
```

> - For _Forms List_ type apps, the form must be in the Forms List menu item.

Author: ```suggestion
- For _Forms List_ type apps, the form can be in the _Forms List_ menu item.
```

- For _Tiles_ type apps, the form can be in a _Container_ menu item or added directly to the menu via a _Form_ menu item.

> - For _Tiles_ or _Forms List_ type apps, the form can be assigned to an action in a scheduled task.

Reviewer 2: ```suggestion
- For _Tiles_ or _Forms List_ type apps, the form can be assigned to an action in a _Scheduled task_.
```

> - For _Approvals_ type apps, the form can be an approval form.

Reviewer 2: ```suggestion
- For _Approvals_ type apps, the form can be an _Approval form_.
```
